examples/rabbit_fox_hawk.jl

```diff
@@ -0,0 +1,417 @@
+        kill_agent!(rand(model.rng, food), model, model.airfinder)
+        hawk.energy += model.Δe_rabbit
+        ## Fly back up
+        set_target!(hawk, hawk.pos .+ (0, 0, 3), model.airfinder)
+    end
+
+    ## The rest of the stepping function is similar to that of foxes, except hawks use a
+    ## different pathfinder
+    hawk.energy -= 1
+    if hawk.energy <= 0
+        kill_agent!(hawk, model, model.airfinder)
+        return
+    end
+
+    rand(model.rng) <= model.hawk_repr && reproduce!(hawk, model)
+
+    if is_stationary(hawk, model.airfinder)
+        prey = [x for x in nearby_agents(hawk, model, model.hawk_vision) if x.type == :rabbit]
+        if isempty(prey)
+            set_target!(
+                hawk,
+                rand(
+                    model.rng,
+                    collect(
+                        nearby_walkable(hawk.pos, model, model.airfinder, model.hawk_vision),
+                    ),
+                ),
+                model.airfinder,
+            )
+        else
+            set_best_target!(hawk, map(x -> x.pos, prey), model.airfinder)
+        end
+    end
+
+    move_along_route!(hawk, model, model.airfinder)
+end
+
+# This function is called when an animal reproduces. The animal loses half its energy, and
+# a copy of it is created and added to the model.
+
+function reproduce!(animal, model)
+    animal.energy = ceil(Int, animal.energy / 2)
+    add_agent_pos!(Animal(nextid(model), animal.pos, animal.type, animal.energy), model)
+end
+
+# The model stepping function simulates the growth of grass
+
+function model_step!(model)
+    ## To prevent copying of data, obtain a view of the part of the grass matrix that
+    ## doesn't have any grass, and grass can grow there
+    growable = view(
+        model.grass,
+        model.grass .== 0 .& model.water_level .< model.heightmap .<= model.grass_level,
+    )
+    ## Grass regrows with a random probability
+    growable .= rand(model.rng, length(growable)) .< model.regrowth_chance
+end
+
+# ## Visualization
+#
+# The agents are color-coded according to their `type`, to make them easily identifiable in
+# the visualization.
+
+using InteractiveDynamics
+using GLMakie
+GLMakie.activate!() # hide
+
+animalcolor(a) =
+    if a.type == :rabbit
+        :brown
+    elseif a.type == :fox
+        :orange
+    else
+        :blue
+    end
+
+# We use `surface!` to plot the terrain as a mesh, and colour it using the `:terrain`
+# colormap. `zlims!` overrides the limits set by `InteractiveDynamics` to ensure the
+# terrain surface isn't skewed by the dimensions of the model.
+function static_preplot!(ax, model)
+    zlims!(ax, (0, 164))
+    surface!(ax, model.heightmap; colormap = :terrain)
+end
+
+## The sample heightmap used for this model
+heightmap_url =
+    "https://raw.githubusercontent.com/JuliaDynamics/" *
+    "JuliaDynamics/master/videos/agents/rabbit_fox_hawk_heightmap.png"
+model = initialize_model(heightmap_url)
+
+abm_video(
```

Comment:
Think limits will still need to be set for the main axis here too. The height seems to be off in the visualization.

Reply:
Also I'd perhaps drop the framerate a little (depending on how the limit fix goes). It's hard to see what's going on at present.

Reply:
The visualization intentionally has a lot of empty space at the top, because that's actually how the terrain should look. Having a non-uniform scale of `(205, 205, 40)` was skewing how everything looked, since the axes were all the same length on the screen but of different scales.

Reply:
Ah. OK. That makes sense. Maybe we can let the eagles fly a little higher. Not ridiculously high, but over the mountain tops a bit to fill some of the dead space.

Reply:
In that case, I'll make it so that eagles can fly over any terrain, instead of being restricted by `mountain_level`

Reply:
This should look a bit better now

Reply:
Tough this one. Slowing it down it's now easier to see that nothing is really happening. I see a lot of rabbits dying but not really why. Will play around a bit with it.

Reply:
Yeah, I've been trying to find a workable combination of parameters too

Reply:
I think they're dying because they were affected the worst by the transition to continuous space. They only roam around randomly, so their chances of stumbling upon food is lower, and their avoidance of predators is also fairly random. It looks like they tend to die a lot after getting corralled into corners and eaten.

Reply:
Let's iterate on the example outside of this PR.